src/Auctions/Auction.sol

```diff
@@ -0,0 +1,652 @@
+     * @return . The price of the auction.
+     */
+    function price(
+        bytes32 _auctionId,
+        uint256 _timestamp
+    ) public view virtual returns (uint256) {
+        // Get unscaled price and scale it down.
+        return
+            _price(
+                auctions[_auctionId].kicked,
+                auctions[_auctionId].initialAvailable *
+                    auctions[_auctionId].fromInfo.scaler,
+                _timestamp
+            ) / wantInfo.scaler;
+    }
+
+    /**
+     * @dev Internal function to calculate the scaled price based on auction parameters.
+     * @param _kicked The timestamp the auction was kicked.
+     * @param _available The initial available amount scaled 1e18.
+     * @param _timestamp The specific timestamp for calculating the price.
+     * @return . The calculated price scaled to 1e18.
+     */
+    function _price(
+        uint256 _kicked,
+        uint256 _available,
+        uint256 _timestamp
+    ) internal view virtual returns (uint256) {
+        if (_available == 0) return 0;
+
+        uint256 secondsElapsed = _timestamp - _kicked;
+
+        if (secondsElapsed > auctionLength) return 0;
+
+        // Exponential decay from https://github.com/ajna-finance/ajna-core/blob/master/src/libraries/helpers/PoolHelper.sol
+        uint256 hoursComponent = 1e27 >> (secondsElapsed / 3600);
+        uint256 minutesComponent = Maths.rpow(
+            MINUTE_HALF_LIFE,
+            (secondsElapsed % 3600) / 60
+        );
+        uint256 initialPrice = Maths.wdiv(startingPrice * 1e18, _available);
+
+        return
+            (initialPrice * Maths.rmul(hoursComponent, minutesComponent)) /
+            1e27;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            SETTERS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Enables a new auction.
+     * @dev Uses governance as the receiver.
+     * @param _from The address of the token to be auctioned.
+     * @return . The unique identifier of the enabled auction.
+     */
+    function enable(address _from) external virtual returns (bytes32) {
+        return enable(_from, msg.sender);
+    }
+
+    /**
+     * @notice Enables a new auction.
+     * @param _from The address of the token to be auctioned.
+     * @param _receiver The address that will receive the funds in the auction.
+     * @return _auctionId The unique identifier of the enabled auction.
+     */
+    function enable(
+        address _from,
+        address _receiver
+    ) public virtual onlyGovernance returns (bytes32 _auctionId) {
+        address _want = want();
+        require(_from != address(0) && _from != _want, "ZERO ADDRESS");
+        require(
+            _receiver != address(0) && _receiver != address(this),
+            "receiver"
+        );
+
+        // Calculate the id.
+        _auctionId = getAuctionId(_from);
```

Comment:
Something I realized on this pass, there isn't something similar to the factory for enabling auction discovery here. Should we also maintain a list of all enabled auction IDs?